overlay/lib/profiles.nix

```diff
@@ -0,0 +1,37 @@
+let
+  profileNames = [
+    "release"
+    "dev"
+    "test"
+    "bench"
+    "__noProfile" # A placeholder profile for build scripts, which don't respect any profiles.
```

Comment:
I understand that `__noProfile` is used only for build scripts, but I believe those are always built in `"dev"` mode. If so, why is this preferable over embedding `profile = "dev"` in the `Cargo.nix` at generation time?

Reply:
Build scripts only inherit (`opt-level` and `debug`)[https://github.com/rust-lang/cargo/blob/rust-1.38.0/src/cargo/core/profiles.rs#L138] from the current profile. I didn't take this into account to simplify things.

Reply:
If they don't make an appreciable difference, could we not select the `"dev"` profile for build scripts in that case?

Reply:
😞 User can set `panic = "abort"` in `[profile.dev]`, which unfortunately isn't compatible with build scripts (which include `build.rs` and proc macros).

Reply:
Ah, makes sense. 😞